### ISSUE TITLE
Running profile checks for training split

### ISSUE THREAD

**Issue by michaelbornholdt** [cytomining/DeepProfiler#256]
I want to run the profile action to extract profiles from a pre-trained net. I have set up a trial with only one plate. 
here is the header to the index.csv 

Unnamed: 0 | Metadata_Plate | Metadata_Well | Metadata_Site | Plate_Map_Name | RNA | ER | AGP | Mito | DNA | Treatment_ID | Treatment_Replicate | Treatment | Compound | Concentration
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | SQ00014812 | A01 | 1 | C-7161-01-LM6-022 | SQ00014812/r01c01f01p01-ch3sk1fk1fl1.tiff | SQ00014812/r01c01f01p01-ch2sk1fk1fl1.tiff | SQ00014812/r01c01f01p01-ch4sk1fk1fl1.tiff | SQ00014812/r01c01f01p01-ch5sk1fk1fl1.tiff | SQ00014812/r01c01f01p01-ch1sk1fk1fl1.tiff | 0 | 1 | DMSO@NA | DMSO | NaN
1 | SQ00014812 | A01 | 2 | C-7161-01-LM6-022 | SQ00014812/r01c01f02p01-ch3sk1fk1fl1.tiff | SQ00014812/r01c01f02p01-ch2sk1fk1fl1.tiff | SQ00014812/r01c01f02p01-ch4sk1fk1fl1.tiff | SQ00014812/r01c01f02p01-ch5sk1fk1fl1.tiff | SQ00014812/r01c01f02p01-ch1sk1fk1fl1.tiff | 0 | 1 | DMSO@NA | DMSO | NaN
2 | SQ00014812 | A01 | 3 | C-7161-01-LM6-022 | SQ00014812/r01c01f03p01-ch3sk1fk1fl1.tiff | SQ00014812/r01c01f03p01-ch2sk1fk1fl1.tiff | SQ00014812/r01c01f03p01-ch4sk1fk1fl1.tiff | SQ00014812/r01c01f03p01-ch5sk1fk1fl1.tiff | SQ00014812/r01c01f03p01-ch1sk1fk1fl1.tiff | 0 | 1 | DMSO@NA | DMSO | NaN
3 | SQ00014812 | A01 | 4 | C-7161-01-LM6-022 | SQ00014812/r01c01f04p01-ch3sk1fk1fl1.tiff | SQ00014812/r01c01f04p01-ch2sk1fk1fl1.tiff | SQ00014812/r01c01f04p01-ch4sk1fk1fl1.tiff | SQ00014812/r01c01f04p01-ch5sk1fk1fl1.tiff | SQ00014812/r01c01f04p01-ch1sk1fk1fl1.tiff | 0 | 1 | DMSO@NA | DMSO | NaN
4 | SQ00014812 | A01 | 5 | C-7161-01-LM6-022 | SQ00014812/r01c01f05p01-ch3sk1fk1fl1.tiff | SQ00014812/r01c01f05p01-ch2sk1fk1fl1.tiff | SQ00014812/r01c01f05p01-ch4sk1fk1fl1.tiff | SQ00014812/r01c01f05p01-ch5sk1fk1fl1.tiff | SQ00014812/r01c01f05p01-ch1sk1fk1fl1.tiff | 0 | 1 | DMSO@NA | DMSO | NaN

Note that I do not have a row for Train and Validation 

Then here is my config.csv
```
{
    "dataset": {
        "metadata": {
            "label_field": "Treatment",
            "control_value": "NA@NA"
        },
        "images": {
            "channels": [
                "DNA",
                "RNA",
                "ER",
                "AGP",
                "Mito"
              ],
            "file_format": "png",
            "bits": 16,
            "width": 696,
            "height": 520
        },
        "locations":{
            "mode": "single_cells",
            "box_size": 128,
            "area_coverage": 0.75,
            "mask_objects": false
        }
    },
    "prepare": {
        "illumination_correction": {
            "down_scale_factor": 4,
            "median_filter_size": 24
        },
        "compression": {
            "implement": true,
            "scaling_factor": 1.0
        }
    },
    "train": {
        "partition": {
            "targets": [
                "Compound"
            ],
            "split_field": "Compound",
            "training_values": ["Training"],
            "validation_values": ["Validation"]
        },
        "model": {
            "name": "efficientnet",
            "crop_generator": "repeat_channel_crop_generator",
            "metrics": ["accuracy", "top_k"],
            "epochs": 100,
            "initialization":"ImageNet",
            "params": {
                "learning_rate": 0.005,
                "batch_size": 64,
                "conv_blocks": 0,
                "feature_dim": 256,
                "pooling": "avg"
            },
            "lr_schedule": "cosine"
        },
        "sampling": {
            "factor": 1,
            "workers": 4,
            "cache_size": 15000
        },
        "validation": {
            "frequency": 2,
            "top_k": 5,
            "batch_size": 32,
            "frame": "val",
            "sample_first_crops": true
        }
    },
    "profile": {
      "use_pretrained_input_size": 224,
      "feature_layer": "avg_pool",
      "checkpoint": "None",
      "batch_size": 8
    }
}
```

## running the following command gives and error: 
```
python3 deepprofiler --root=/home/ubuntu/work/test_dp/ --config config.json profile
```
## Error 
```
Reading metadata form /home/ubuntu/work/test_dp//inputs/metadata/index.csv
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3456 entries, 0 to 3455
Data columns (total 15 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Unnamed: 0           3456 non-null   int64 
 1   Metadata_Plate       3456 non-null   object
 2   Metadata_Well        3456 non-null   object
 3   Metadata_Site        3456 non-null   int64 
 4   Plate_Map_Name       3456 non-null   object
 5   RNA                  3456 non-null   object
 6   ER                   3456 non-null   object
 7   AGP                  3456 non-null   object
 8   Mito                 3456 non-null   object
 9   DNA                  3456 non-null   object
 10  Treatment_ID         3456 non-null   int64 
 11  Treatment_Replicate  3456 non-null   int64 
 12  Treatment            3456 non-null   object
 13  Compound             3456 non-null   object
 14  Concentration        3456 non-null   object
dtypes: int64(4), object(11)
memory usage: 405.1+ KB
None
{'BRD-A20131130-001-01-7': 0, 'BRD-A36010170-001-04-0': 1, 'BRD-A49838158-001-01-1': 2, 'BRD-A50684349-003-02-9': 3, 'BRD-A68009927-003-03-7': 4, 'BRD-A87983072-001-01-1': 5, 'BRD-A89585551-003-16-6': 6, 'BRD-K03353589-236-01-4': 7, 'BRD-K31339321-001-03-6': 8, 'BRD-K31928526-001-02-1': 9, 'BRD-K32318651-001-23-7': 10, 'BRD-K33251802-001-01-3': 11, 'BRD-K34008116-001-01-7': 12, 'BRD-K34415467-003-13-0': 13, 'BRD-K35007173-001-07-5': 14, 'BRD-K35960502-001-20-0': 15, 'BRD-K36269323-300-02-6': 16, 'BRD-K36270037-001-01-7': 17, 'BRD-K36740062-001-06-6': 18, 'BRD-K36862742-001-25-6': 19, 'BRD-K36927236-001-32-6': 20, 'BRD-K36970462-001-01-6': 21, 'BRD-K37516142-001-10-5': 22, 'BRD-K47539947-001-03-7': 23, 'BRD-K50691590-001-02-2': 24, 'BRD-K58550667-001-08-7': 25, 'BRD-K60038276-001-09-0': 26, 'BRD-K60230970-001-10-0': 27, 'BRD-K63712959-001-01-8': 28, 'BRD-K69688083-004-23-1': 29, 'BRD-K73691377-001-01-6': 30, 'BRD-K75024268-001-01-0': 31, 'BRD-K76703230-004-04-1': 32, 'BRD-K77771411-003-10-7': 33, 'BRD-K80535353-001-01-4': 34, 'BRD-K82164249-001-03-5': 35, 'BRD-K87202646-061-04-9': 36, 'BRD-K87278688-001-01-0': 37, 'BRD-K87990216-001-10-2': 38, 'BRD-K88025533-003-03-3': 39, 'BRD-K88052444-001-08-8': 40, 'BRD-K88789588-001-14-9': 41, 'BRD-K88807631-001-01-3': 42, 'BRD-K89014967-001-04-3': 43, 'BRD-K89015388-001-01-7': 44, 'BRD-K89104321-003-14-8': 45, 'BRD-K89402695-001-02-3': 46, 'BRD-K89561498-001-01-7': 47, 'BRD-K89704198-001-12-3': 48, 'BRD-K90239174-001-01-8': 49, 'BRD-K90789829-001-07-8': 50, 'BRD-K91290917-300-01-8': 51, 'BRD-K92731339-227-04-9': 52, 'BRD-K96029467-003-01-9': 53, 'BRD-K97688263-001-03-7': 54, 'BRD-K98453471-001-01-7': 55, 'BRD-K99107520-001-24-1': 56, 'BRD-M98279124-300-01-1': 57, 'DMSO': 58}
Reading single-cell locations
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "deepprofiler/__main__.py", line 197, in <module>
    cli(obj={})
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "deepprofiler/__main__.py", line 180, in profile
    metadata = deepprofiler.dataset.image_dataset.read_dataset(context.obj["config"])
  File "/home/ubuntu/work/DeepProfiler/deepprofiler/dataset/image_dataset.py", line 239, in read_dataset
    dset.prepare_training_locations()
  File "/home/ubuntu/work/DeepProfiler/deepprofiler/dataset/image_dataset.py", line 74, in prepare_training_locations
    locations = pd.concat(locations)
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 284, in concat
    sort=sort,
  File "/home/ubuntu/miniconda3/envs/tf/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 331, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate

```

**Comment by michaelbornholdt** [comment]
@jccaicedo @Arkkienkeli 
Any ideas what is going on or what I should be changing? 

in general, I think when running profile - nothing with the training should be touched

**Comment by jccaicedo** [comment]
Seems like the problem you are having is with the location files. Are these properly configured and named?

**Comment by michaelbornholdt** [comment]
I have solved the problem by creating a dummy column named split. I just filled the whole column with 'training' 
But surely this is not the intended behaviors since I am running profile.

**Comment by Arkkienkeli** [comment]
Probably we could ignore the `partition` section during the profiling as we don't need it. 
I will look into this next week.

**Comment by Arkkienkeli** [comment]
Fixed with e23a66a6474dd8d0cc0c299a45e39a6a8261c7c8
